etc/vcl_snippets/deliver.vcl

```diff
@@ -17,6 +16,23 @@
         # Remove X-Magento-Vary and HTTPs Vary served to the user
         set resp.http.Vary = regsub(resp.http.Vary, "X-Magento-Vary,Https", "Cookie");
         remove resp.http.X-Magento-Tags;
+
+        # Set headers based on suggestions in https://www.fastly.com/blog/headers-we-want
+        set resp.http.Content-Security-Policy = "default-src 'self'; frame-ancestors 'self'";
+        set resp.http.Referer-Policy = "origin-when-cross-origin";
+        set resp.http.X-XSS-Protection = "1; mode=block";
+        set resp.http.X-Content-Type-Options = "nosniff";
+
+        # CORS policy. Sets them only if they have not been set already
```

Comment:
Can we get clarification on why we should add them?  According to the blog, https://www.fastly.com/blog/headers-we-want , "These headers are permissive, not restrictive, so not having them at all gives you the highest level of security."

Reply:
Agreed, don't set these unless the site is doing cross origin API calls from the browser